tests/functional/lib/helpers.js

```diff
@@ -4,135 +4,49 @@
-          if (url.includes("return_to")) {
-            return (
-              context.remote
-                .findByCssSelector("#login_field")
-                .click()
-                .type(config.wc.user)
-                .end()
-                .findByCssSelector("#password")
-                .click()
-                .type(config.wc.pw)
-                .end()
-                .findByCssSelector("input[type=submit]")
-                .click()
-                .end()
-                // *Sometimes* GitHub can bring up an extra verification
-                // page if it detects that our test user is requesting
-                // access too much.
-                .findByCssSelector(".oauth-review-permissions")
-                .then(
-                  function() {
-                    // In this case, there's an extra button to click to convince
-                    // GitHub we're totally not a bot. >_>
-                    return context.remote
-                      .sleep(3000)
-                      .findByCssSelector("button.btn-primary")
-                      .click()
-                      .end();
-                  },
-                  function(err) {
-                    // Otherwise, we swallow the NoSuchElement error.
-                    return new Promise(function(resolve) {
-                      if (/NoSuchElement/.test(String(err))) {
-                        resolve(true);
-                      } else {
-                        throw err;
-                      }
-                    });
-                  }
-                )
-                .end()
-                // ...Now make sure the logged-in avatar is shown so we know we're
-                // back at the home page before we end.
-                .findByCssSelector(".wc-Navbar-avatar")
-                .end()
-            );
-          }
-        })
-        .end();
-    }
-
-    function logout(context) {
-      return openPage(context, url("/logout"), "body").clearCookies().end();
+  async function openPage(context, url, readySelector, longerTimeout) {
+    let timeout = longerTimeout ? 20000 : config.wc.pageLoadTimeout;
+    let remote = context.remote;
+    try {
+      await remote.get(require.toUrl(url));
+      await remote.setFindTimeout(timeout);
+      await remote.findByCssSelector(readySelector);
+    } catch (error) {
+      let url = await remote.getCurrentUrl();
+      console.log("Error fetching %s", url);
+      throw new Error(error);
     }
+  }
 
-    function visibleByClass(selector) {
-      var elem;
-      return pollUntil(
-        function(selector) {
-          elem = document.getElementsByClassName(selector);
-          if (!elem || elem.length === 0) {
-            return null;
-          }
-          elem = elem[0];
-          return elem.offsetWidth > 0 && elem.offsetHeight > 0 ? true : null;
-        },
-        [selector],
-        10000
-      );
-    }
+  async function login(context) {
```

Comment:
Ah, yes! 

I removed that just to make the transition faster for me, since this function was not the one that I actually wanted to test. 

But now I'm thinking, since we have the mock login in place is that really needed?